### DISCUSSION TITLE
How to connect to speakers

### DISCUSSION THREAD

**Discussion by daniel-sousa-me** [CarlosDerSeher/snapclient#29] [General]
I've been using some Raspberry Pis to have multiroom audio, but I also have some esp32 dev kits lying around. It would be nice to replace the Pis with stuff that costs 1/10

I have very little experience with microcontrollers and esp. Can I just flash this and output it to some 3.5 audio jack? Or do I need an external DAC?

My amplifiers have Bluetooth so they need to have some kind of DAC, but it's annoying that they don't have any other digital input!

**Comment by CarlosDerSeher** [comment]
Check #27 there you'll find some information about supported DACs. Connect DAC through I2S to ESP32. Normally DAC has an analog out through an audio jack which then can be connected to your amplifiers line in.

**Reply by daniel-sousa-me** [reply]
Thank you very much! For the pointers, but also for the code!

**Reply by CarlosDerSeher** [reply]
Feel free to also try the develop branch if you want to test opus and pcm codec support.

**Reply by daniel-sousa-me** [reply]
What are the advantages? From the little I understand, flac seems to be the best option

**Reply by CarlosDerSeher** [reply]
I agree, personally I use FLAC only. I just need people testing the code to reveal any issues :)
